analyzer/src/main/kotlin/managers/utils/MavenDependencyHandler.kt

```diff
@@ -78,11 +84,23 @@ class MavenDependencyHandler(
     override fun linkageFor(dependency: DependencyNode): PackageLinkage =
         if (isLocalProject(dependency)) PackageLinkage.PROJECT_DYNAMIC else PackageLinkage.DYNAMIC
 
+    /**
+     * Create a [Package] representing [dependency] if possible, adding potential issues to [issues]. Inter-project
+     * dependencies are skipped. There is the corner case that a dependency references a project, but in a different
+     * version than the one used by the local build. Then, this dependency is actually a package, but Maven's
+     * resolution mechanism might prefer using the project. Therefore, the check whether [dependency] is a project
+     * must be done before and after the package resolution.
+     */
     override fun createPackage(dependency: DependencyNode, issues: MutableList<OrtIssue>): Package? {
         if (isLocalProject(dependency)) return null
 
         return runCatching {
-            support.parsePackage(dependency.artifact, dependency.repositories, localProjects, sbtMode)
+            support.parsePackage(dependency.artifact, dependency.repositories, sbtMode = sbtMode).let { pkg ->
+                if (isLocalProject(pkg.id)) {
+                    localProjectIds += dependency.identifier()
+                    null
+                } else pkg
```

Comment:
Nit: When `if` is a block, `else` should be a block, too, for symmetry / readability.